### PR TITLE
Remove `T` to convert vector

### PR DESCRIPTION
**Pull request by tmigot**
#126 

```
using ADNLPModels, OptimizationProblems, NLPModels
ad_problems = [OptimizationProblems.ADNLPProblems.eval(Symbol(pb))() for pb in setdiff(OptimizationProblems.meta[!, :name], ["hs110"])]
for pb in ad_problems @show pb.meta.name;x = rand(pb.meta.nvar);jac(pb, x);grad(pb, x) end
```
now works.

### CODE REVIEW

File: src/ADNLPProblems/AMPGO13.jl [hunk 1]
```diff
@@ -3,7 +3,7 @@ export AMPGO13
 function AMPGO13(; n::Int = default_nvar, type::Val{T} = Val(Float64), kwargs...) where {T}
   function f(x)
     n = length(x)
-    return 0 < x[1] < 1 ? -T(x[1]^(2 / 3) + (1 - x[1]^2)^(1 / 3)) : T[Inf]
+    return 0 < x[1] < 1 ? -(x[1]^T(2 / 3) + (1 - x[1]^2)^T(1 / 3)) : T[Inf]
```
Comment (dpo) [inline src/ADNLPProblems/AMPGO13.jl:6]: I don't think it's necessary to enclose Ints or rationals with `T` either. Also `T[Inf]` could again cause problem. Is there some sort of `convert(eltype(x), Inf)`?

File: src/ADNLPProblems/hs10.jl [hunk 2]
```diff
@@ -3,7 +3,7 @@ export hs10
 function hs10(; n::Int = default_nvar, type::Val{T} = Val(Float64), kwargs...) where {T}
   x0 = T[-10; 10]
   f(x) = x[1] - x[2]
-  c(x) = T[-3 * x[1]^2 + 2 * x[1] * x[2] - x[2]^2 + 1]
+  c(x) = [-3 * x[1]^2 + 2 * x[1] * x[2] - x[2]^2 + 1]
   lcon = T[0.0]
```
Comment (dpo) [inline src/ADNLPProblems/hs10.jl:7]: explicit Float64 here

Comment (tmigot) [reply src/ADNLPProblems/hs10.jl:7]: There is a unit test for this and the type seems correct (not restricted to Float64).

File: src/ADNLPProblems/AMPGO13.jl (hunk 1 above)
Comment (abelsiqueira) [reply src/ADNLPProblems/AMPGO13.jl:6]: Also `T(1 / 3)` will create Float64 1/3 and then convert to `T`, which for `BigFloat` is loss of precisions. `(T(2) / 3)` is uglier but better.

File: src/ADNLPProblems/hs25.jl [hunk 3]
```diff
@@ -1,10 +1,10 @@
 export hs25
 
 function hs25(; n::Int = default_nvar, type::Val{T} = Val(Float64), kwargs...) where {T}
-  u = [25 + (-50 * log(0.01 * i))^(2 / 3) for i = 1:99]
+  u = T[25 + (-50 * log(0.01 * i))^(2 / 3) for i = 1:99]
```
Comment (abelsiqueira) [inline src/ADNLPProblems/hs25.jl:4]: Same comment as earlier

File: src/ADNLPProblems/AMPGO13.jl (hunk 1 above)
Comment (dpo) [reply src/ADNLPProblems/AMPGO13.jl:6]: I think we're supposed to leave it  as `2 / 3` to be generic.

Comment (abelsiqueira) [reply src/ADNLPProblems/AMPGO13.jl:6]: What do you mean by generic?

Comment (dpo) [reply src/ADNLPProblems/AMPGO13.jl:6]: I mean that Ints and rationals will be converted automatically to the appropriate floating-point type.

Comment (abelsiqueira) [reply src/ADNLPProblems/AMPGO13.jl:6]: `2 / 3` is just the `Int64` division. We can use `2 // 3` instead to have Rational, but that might be slower.

Comment (dpo) [reply src/ADNLPProblems/AMPGO13.jl:6]: No actually 2/3 is a Float64. Let's use 2//3 then.

**Review by abelsiqueira — APPROVED** [review]
LGTM, thanks.

**Pull request merged**